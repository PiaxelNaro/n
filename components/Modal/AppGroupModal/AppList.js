// @flow
import React from 'react';
import styled from 'styled-components';
import FlexGrid from '../../FlexGrid';
import AppIcon from '../../AppIcon';
import { sprintf, type I18nType } from '../../../modules/i18n';

const MAX_DISPLAY_APP_COUNT = 10;
const Wrapper = styled.div`
    border-bottom: 1px solid #F2F2F2;
`;

const Notice = styled.p`
    margin: 10px 0;
    color: #102235;
`;

const AppList = ({
    apps,
    i18n,
}: {
    apps: Array<{ id: number, icon: string, store: string }>,
    i18n: I18nType,
}) => {
    const { gettext } = i18n;
    let displayedApps = apps;
    let moreApps = null;
    if (apps.length > MAX_DISPLAY_APP_COUNT) {
        const displayCount = MAX_DISPLAY_APP_COUNT - 1;
        displayedApps = apps.slice(0, displayCount);
        moreApps = (
            <span className="js-more-apps-marker">
                {sprintf(gettext('+%d more'), apps.length - displayCount)}
            </span>
        );
    }

    const selectedApps = displayedApps.map(app =>
        <AppIcon key={app.id} src={app.icon} store={app.store} size="medium" />
    );

    return (
        <Wrapper>
            <FlexGrid gutter="10px">
                {selectedApps}
                {moreApps}
            </FlexGrid>
            <Notice>
                {gettext(
                    'We will automatically add selected apps across platforms and devices.'
                )}
            </Notice>
        </Wrapper>
    );
};

export default AppList;
