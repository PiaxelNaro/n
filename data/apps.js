export default [
    {
        id: 1,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80',
        company: 'Networkcanal Multimedia Ltd',
        name: 'Fa',
    },
    {
        id: 2,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80',
        company: 'KAPS Development',
        name: 'FA',
    },
    {
        id: 3,
        icon:
            'https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80',
        company: 'INNERGY CO.,LTD',
        name: '와이드앵글 FA 교육',
    },
    {
        id: 4,
        icon:
            'https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80',
        company: '7700 Software',
        name: 'FA World',
    },
    {
        id: 5,
        icon:
            'https://static-s.aa-cdn.net/img/ios/688824991/b3ca3dbc0fb23363af7aa7917d4728a8_w80',
        company: 'Lubomir Havran BINARTS',
        name: 'FA Magazín',
    },
    {
        id: 6,
        icon:
            'https://static-s.aa-cdn.net/img/ios/805530850/d5b350ceb68604faafe6a3c234b07bb7_w80',
        company: 'Tobit',
        name: 'FA/KE',
    },
    {
        id: 7,
        icon:
            'https://static-s.aa-cdn.net/img/ios/884096407/1f6d3ba08a891906570ad2bb68b8dcdd_w80',
        company: 'Tobit',
        name: 'Fa. Handelsagent',
    },
    {
        id: 8,
        icon:
            'https://static-s.aa-cdn.net/img/ios/892841769/b27b0d7ccf10c6182fcf6ddf30017421_w80',
        company: 'Spinn Labs Private Limited',
        name: 'Bao Fa',
    },
    {
        id: 9,
        icon:
            'https://static-s.aa-cdn.net/img/ios/906627580/f41757b0e6c2509e20a937e80e2e0233_w80',
        company: 'E Sports Publications Limited',
        name: 'Lincolnshire FA',
    },
    {
        id: 10,
        icon:
            'https://static-s.aa-cdn.net/img/ios/919412577/ad1ae851ddf94c080bc93ad41c0d20fe_w80',
        company: 'Tobit',
        name: 'FA Foto',
    },
    {
        id: 11,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80',
        company: 'Networkcanal Multimedia Ltd',
        name: 'Fa1',
    },
    {
        id: 12,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80',
        company: 'KAPS Development',
        name: 'FA1',
    },
    {
        id: 13,
        icon:
            'https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80',
        company: 'INNERGY CO.,LTD',
        name: '와이드앵글 FA 교육1',
    },
    {
        id: 14,
        icon:
            'https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80',
        company: '7700 Software',
        name: 'FA World1',
    },
];
