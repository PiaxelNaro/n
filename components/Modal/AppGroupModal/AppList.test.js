// @flow
import React from 'react';
import { shallow } from 'enzyme';
import AppList from './AppList';
import apps from '../../../data/apps';
import mockI18n from '../../../modules/i18n/mockI18n';

describe('<AppList>', () => {
    it('test appList', () => {
        let appList = shallow(<AppList apps={apps} i18n={mockI18n} />);
        expect(appList).toMatchSnapshot();
        expect(appList.find('AppIcon').length).toBe(9);
        expect(appList.find('.js-more-apps-marker').length).toBe(1);

        appList = shallow(<AppList apps={apps.slice(0, 5)} i18n={mockI18n} />);
        expect(appList).toMatchSnapshot();
        expect(appList.find('AppIcon').length).toBe(5);
        expect(appList.find('.js-more-apps-marker').length).toBe(0);
    });
});
