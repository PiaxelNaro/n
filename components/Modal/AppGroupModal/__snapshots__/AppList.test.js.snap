// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppList> test appList 1`] = `
<AppList__Wrapper>
  <FlexGrid
    gutter="10px"
  >
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/688824991/b3ca3dbc0fb23363af7aa7917d4728a8_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/805530850/d5b350ceb68604faafe6a3c234b07bb7_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/884096407/1f6d3ba08a891906570ad2bb68b8dcdd_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/892841769/b27b0d7ccf10c6182fcf6ddf30017421_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/906627580/f41757b0e6c2509e20a937e80e2e0233_w80"
    />
    <span
      className="js-more-apps-marker"
    >
      +5 more
    </span>
  </FlexGrid>
  <AppList__Notice>
    We will automatically add selected apps across platforms and devices.
  </AppList__Notice>
</AppList__Wrapper>
`;

exports[`<AppList> test appList 2`] = `
<AppList__Wrapper>
  <FlexGrid
    gutter="10px"
  >
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80"
    />
    <AppIcon
      size="medium"
      src="https://static-s.aa-cdn.net/img/ios/688824991/b3ca3dbc0fb23363af7aa7917d4728a8_w80"
    />
  </FlexGrid>
  <AppList__Notice>
    We will automatically add selected apps across platforms and devices.
  </AppList__Notice>
</AppList__Wrapper>
`;
